fastseq_cli/transformers_generate.py

```diff
@@ -5,9 +5,78 @@
+}))
+
+transformers.CONFIG_MAPPING.update(dict({
+    "unilm": UnilmConfig
+}))
+
+class AutoModelForSeq2SeqLM(_AutoModelForSeq2SeqLM):
+    @classmethod
+    def from_pretrained(cls, pretrained_model_name_or_path, *model_args, **kwargs):
+        config = kwargs.pop("config", None)
+        if not isinstance(config, PretrainedConfig):
+            if pretrained_model_name_or_path in UNILM_PRETRAINED_CONFIG_ARCHIVE_MAP:
+                pretrained_config_name_or_path = UNILM_PRETRAINED_CONFIG_ARCHIVE_MAP[pretrained_model_name_or_path]
+            else:
+                pretrained_config_name_or_path = pretrained_model_name_or_path
+            config = AutoConfig.from_pretrained(pretrained_config_name_or_path, **kwargs)
+
+        for config_class, model_class in transformers.MODEL_FOR_SEQ_TO_SEQ_CAUSAL_LM_MAPPING.items():
+            if isinstance(config, config_class):
+                return model_class.from_pretrained(pretrained_model_name_or_path, *model_args, config=config, **kwargs)
+        raise ValueError(
+            "Unrecognized configuration class {} for this kind of AutoModel: {}.\n"
+            "Model type should be one of {}.".format(
+                config.__class__,
+                cls.__name__,
+                ", ".join(c.__name__ for c in transformers.MODEL_FOR_SEQ_TO_SEQ_CAUSAL_LM_MAPPING.keys()),
+            )
+        )
+
+class AutoTokenizer(_AutoTokenizer):
+    @classmethod
+    def from_pretrained(cls, pretrained_model_name_or_path, *inputs, **kwargs):
+        config = kwargs.pop("config", None)
+        if not isinstance(config, PretrainedConfig):
+            if pretrained_model_name_or_path in UNILM_PRETRAINED_CONFIG_ARCHIVE_MAP:
+                pretrained_config_name_or_path = UNILM_PRETRAINED_CONFIG_ARCHIVE_MAP[pretrained_model_name_or_path]
+            else:
+                pretrained_config_name_or_path = pretrained_model_name_or_path
+            config = AutoConfig.from_pretrained(pretrained_config_name_or_path, **kwargs)
+
+        if "bert-base-japanese" in str(pretrained_model_name_or_path):
+            return BertJapaneseTokenizer.from_pretrained(pretrained_model_name_or_path, *inputs, **kwargs)
+
+        use_fast = kwargs.pop("use_fast", False)
+        for config_class, (tokenizer_class_py, tokenizer_class_fast) in transformers.TOKENIZER_MAPPING.items():
+            if isinstance(config, config_class):
+                if tokenizer_class_fast and use_fast:
+                    return tokenizer_class_fast.from_pretrained(pretrained_model_name_or_path, *inputs, **kwargs)
+                else:
+                    return tokenizer_class_py.from_pretrained(pretrained_model_name_or_path, *inputs, **kwargs)
+
+        raise ValueError(
+            "Unrecognized configuration class {} to build an AutoTokenizer.\n"
+            "Model type should be one of {}.".format(
+                config.__class__, ", ".join(c.__name__ for c in TOKENIZER_MAPPING.keys())
+            )
+        )
```

Comment:
If users want to use `AutoModelForSeq2SeqLM` to load UniLM, they need to use this class instead of the one in huggingface, right? If yes, could we use `@replace` decorator to replace the ones in huggingface? So that no matter which one users import, they can always load UniLM. And these two classes can be moved to `fastseq/models/modeling_auto_hf.py`